### ISSUE TITLE
Difference between dy.LSTMBuilder with tensorflow tf.nn.static_rnn/dynamic_rnn

### ISSUE THREAD

**Issue by zjuwfz** [clab/dynet#1508]
I'm doing a nlp experiment using the bilstm-crf model with tensorflow, and I find the result of others with dynet is better than mine though the structure and params of the model is almost the same. I wonder if the implementation of LSTM result in it. And what's the difference between dy.LSTMBuilder with tensorflow tf.nn.static_rnn/dynamic_rnn? For example, tensorflow needs to do padding first while dynet doesn't need to do padding. How does dynet deal with sequences of different lengths?

**Comment by philip30** [comment]
Hi, you might want to see the article here: https://medium.com/@Petuum/intro-to-dynamic-neural-networks-and-dynet-67694b18cb23 about static vs dynamic declaration.
